pkg/parse/events/publisher.go

```diff
@@ -201,6 +204,82 @@ type ResetOnRunAttemptPublisher struct {
 func (s *ResetOnRunAttemptPublisher) HandleResult(result Result) {
 	s.TimeDelayPublisher.HandleResult(result)
 	if result.RunAttempted {
+		klog.V(3).Infof("Run attempted, delaying next sync event (%v)", s.Period)
 		s.timer.Reset(s.Period)
 	}
 }
+
+// NewDedupingChannelPublisher constructs an ChannelDrivenPublisher that
+// enqueues a new specified event any time the input channel receives an event.
+func NewDedupingChannelPublisher(eventType EventType, inCh <-chan bool) *ChannelDrivenPublisher {
```

Comment:
It doesn't send a RetrySyncEventType. It sends a SyncEventType, which is passed in to the constructor by the builder.

This does mess up the triggers used for logging a bit, but it should work fine, because the handler does basically the same thing for both events now.

Reply:
My ultimate goal is to condense all the events down to just one type. I have a different PR to combine the Sync and Reimport events. So this one moves the conflict and watch update cases to use the Sync event too.